File: frontend/src/types/project.ts
export type SourceType = "text" | "pdf" | "docx" | "document" | "image" | "url" | "video";


export type OutputType = "executive_summary" | "advisory" | "linkedin_post" | "video_package";

export interface SourceClaim {
  claim: string;
  source_evidence: string;
}

export interface ExecutiveSummary {
  title: string;
  summary: string;
  key_points: string[];
  recommended_actions: string[];
  source_claims: SourceClaim[];
}

export interface Advisory {
  title: string;
  issued_date: string;
  audience: string;
  situation: string;
  impact: string;
  actions: string[];
  reporting_guidance: string;
  source_claims: SourceClaim[];
}

export interface LinkedInPost {
  headline: string;
  body: string;
  hashtags: string[];
  call_to_action: string;
  source_claims: SourceClaim[];
}

export interface VideoScene {
  scene_number: number;
  duration_seconds: number;
  visual: string;
  narration: string;
}

export interface VideoPackage {
  title: string;
  duration_seconds: number;
  creative_concept: string;
  scenes: VideoScene[];
  narration: string;
  subtitles: string[];
  visual_recommendations: string[];
  source_claims: SourceClaim[];
}

export interface GenerationVersion {
  id: string;
  generation_id: string;
  content: any;
  version_number: number;
  created_at: string;
}

export interface Generation {
  id: string;
  project_id: string;
  output_type: OutputType;
  content: any;
  model: string;
  status: "ready" | "generating" | "failed";
  created_at: string;
  updated_at?: string;
  versions?: GenerationVersion[];
}

export interface SourceAsset {
  id: string;
  project_id: string;
  filename: string;
  mime_type: string;
  storage_path?: string;
  extracted_text?: string;
  extraction_status: "pending" | "completed" | "failed";
  created_at: string;
}

export interface GenerationRequest {
  output_types: OutputType[];
  audience: string;
  tone: string;
  objective: string;
  detail_level: "concise" | "standard" | "detailed";
  language: string;
}

export interface Project {
  id: string;
  title: string;
  source_type: SourceType;
  source_text?: string | null;
  settings: Record<string, any>;
  status: "draft" | "extracting" | "ready" | "generating" | "completed" | "failed";
  created_at: string;
  updated_at: string | null;
  assets?: SourceAsset[];
  generations?: Generation[];
}

export interface OperatorProfile {
  id: string;
  name: string;
  badge: string;
  description: string;
  defaultAudience: string;
  defaultTone: string;
  defaultObjective: string;
  defaultDetailLevel: "concise" | "standard" | "detailed";
  defaultOutputs: OutputType[];
}

File: backend/app/schemas/project.py
from datetime import datetime
from typing import Any
from pydantic import BaseModel, ConfigDict, Field
from app.schemas.generation import GenerationRead


class SourceAssetRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    project_id: str
    filename: str
    mime_type: str
    storage_path: str | None = None
    extracted_text: str | None = None
    extraction_status: str
    created_at: datetime


class ProjectCreate(BaseModel):
    title: str = Field(min_length=1, max_length=140)
    source_type: str = Field(default="text", max_length=32)
    source_text: str | None = None
    settings: dict[str, Any] = Field(default_factory=dict)


class ProjectUpdate(BaseModel):
    title: str | None = None
    source_text: str | None = None
    settings: dict[str, Any] | None = None
    status: str | None = None


class ProjectRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    title: str
    source_type: str
    source_text: str | None = None
    settings: dict[str, Any] = Field(default_factory=dict)
    status: str
    created_at: datetime
    updated_at: datetime | None = None
    assets: list[SourceAssetRead] = []
    generations: list[GenerationRead] = []
